models/TeacherDetailModel.js

```diff
@@ -0,0 +1,136 @@
+const mongoose = require('mongoose');
+const crypto = require('crypto');
+const validator = require('validator');
+const bcrypt = require('bcryptjs');
+
+const teacherDetailSchema = new mongoose.Schema({
+    tid: {
+        type: String,
+        require: [true, 'Please enter your Id'],
+        unique: true,
+        trim: true,
+        lowercase: true
+    },
+
+    name: {
+        type: String,
+        require: true,
+        trim: true,
+    },
+
+    age: {
+        type: Number,
+        require: true,
+        max: [100, 'Age are not allowed greater than 100']
+    },
+
+    gender: {
+        type: String,

```

Comment:
Here give your own set of allowed gender categories, so that no one can enter some random string.

Reply:
okay